### PR TITLE
Fix base64 detection

### LINKED ISSUE #15: Better recognition if base64 decoding of token is needed
If providing plain old access token, for example GitHub personal access token, `Base64.strict_decode64` does not raise exception.

### PR DESCRIPTION
**Pull request by jnummelin**
fixes #15 

@nevalla could you try it out with your problematic token

### CODE REVIEW

**Comment by jnummelin** [comment]
Seems that it's pretty much impossible to 100% accurately detect if a string is base64 or not.

**Comment by jnummelin** [comment]
Changed this now to always expect to see `KUBE_TOKEN` as base64 encoded. If not able to decode, it shows error and exits. PTAL @nevalla

File: lib/mortar/command.rb [hunk 1]
```diff
@@ -153,12 +151,21 @@ def build_kubeconfig_from_env
         preferences: {},
         current_context: 'mortar'
       )
+    rescue ArgumentError
+      puts "KUBE_TOKEN env doesn't seem to be valid base64 encoding!"
```
Comment (kke) [inline lib/mortar/command.rb]: something like "doesn't seem to be base64 encoded"

also you can use `signal_usage_error "..."` or `abort "..."` and won't need to exit with error separately.

File: lib/mortar/command.rb [hunk 2]
```diff
@@ -153,12 +151,21 @@ def build_kubeconfig_from_env
...
     # Stringifies all hash keys
     # @return [Hash]
     def stringify_hash(hash)
       JSON.load(JSON.dump(hash))
     end
+
+    # Detects if a given string is base64 encoded or not
+    # @return [Boolean]
+    def base64?(str)
+      Base64.encode64(Base64.decode64(str)) == str
```
Comment (kke) [inline lib/mortar/command.rb]: i think this has potential to fail depending on if the str being strict or not.

File: lib/mortar/command.rb (hunk 1 above)
Comment (kke) [reply lib/mortar/command.rb]: and you could do the option processing in the option block:

```ruby
option '--kube-token`, 'TOKEN', 'Kubernetes API access token', environment_variable: 'KUBE_TOKEN' do |token|
  begin
    Base64.decode64(token).tap do |decoded|
      raise ArgumentError unless Base64.encode64(decoded) == token
    end
  rescue ArgumentError
    signal_usage_error 'kube token does not seem to be base64 encoded'
  end
end
```

i think the same should be done for `KUBE_CA` and `KUBE_SERVER` too to document them in the `--help` output (and add a secondary way to pass them in)

but maybe that's a separate pr.

File: lib/mortar/command.rb (hunk 2 above)
Comment (kke) [reply lib/mortar/command.rb]: Oh but we don't accept anything but strict.

File: lib/mortar/command.rb [hunk 3]
```diff
@@ -153,6 +151,8 @@ def build_kubeconfig_from_env
         preferences: {},
         current_context: 'mortar'
       )
+    rescue ArgumentError
+      signal_usage_error "KUBE_TOKEN env doesn't seem to be valid base64 encoding!"
```
Comment (kke) [inline lib/mortar/command.rb]: "doesn't seem to be valid base64 encoding" still sounds wrong.

Comment (jnummelin) [reply lib/mortar/command.rb]: changed to `KUBE_TOKEN env doesn't seem to be base64 encoded!`, ok now?

Comment (kke) [reply lib/mortar/command.rb]: yes :)

**Comment by jnummelin** [comment]
@kke review status?

**Pull request merged**